lib/ace/ext/papeeria/context-previewer.coffee

```diff
@@ -0,0 +1,133 @@
+define((require, exports, module) ->
+  exports.setupPreviewer = (editor, popoverHandler) ->
+    katex = null
+    popoverHandler = popoverHandler ? {
+      options: {
+        html: true
+        placement: "bottom"
+        trigger: "manual"
+        title: "Formula"
+        container: editor.container
+      }
+
+      show: (jqPopoverContainer, content, position) ->
+        jqPopoverContainer.css(position)
+        popoverHandler.options.content = content
+        jqPopoverContainer.popover(popoverHandler.options)
+        jqPopoverContainer.popover("show")
+        return
+
+      destroy: (jqPopoverContainer) ->
+        jqPopoverContainer.popover("destroy")
+
+      isVisible: (jqPopoverContainer) ->
+        jqPopoverContainer.data().popover.tip().hasClass("in")
+
+      popoverExists: (jqPopoverContainer) ->
+        jqPopoverContainer.data()? and jqPopoverContainer.data().popover?
+
+      setContent: (jqPopoverContainer, content, position) ->
+        jqPopoverElement = jqPopoverContainer.data().popover.tip().children(".popover-content")
+        jqPopoverElement.html(content)
+    }
+
+    initKaTeX = (onLoaded) ->
```

Comment:
Well katex-previewer works a bit differently. Are you saying I should have just replaced the katex-previewer functionality?